lib/chef/knife/ec2_server_create.rb

```diff
@@ -507,6 +507,9 @@ def bootstrap_common_params(bootstrap)
         bootstrap.config[:encrypted_data_bag_secret_file] = locate_config_value(:encrypted_data_bag_secret_file)
         bootstrap.config[:secret] = s3_secret || locate_config_value(:secret)
         bootstrap.config[:secret_file] = locate_config_value(:secret_file)
+        unless locate_config_value(:encrypted_data_bag_secret_file)
```

Comment:
`bootstrap.config[:encrypted_data_bag_secret_file]` has been referred at line#507 too. We can have a single `If-Else` condition for this.